File: test-fixtures/cross-language/php/Fields.php
<?php

class Ledger {
    public int $mutableField;
    public readonly int $immutableField;
    public static int $staticField;
    private string $privateField;
    public int $sharedName;

    public function __construct(public int $promotedField) {
        $this->mutableField = $promotedField;
        $this->immutableField = 1;
        $this->privateField = "";
    }
}

class Archive {
    public int $sharedName;
}

File: test-fixtures/cross-language/ruby/fields.rb
class Ledger
  # @return [Integer]
  attr_accessor :mutable_field
  # @return [Integer]
  attr_reader :immutable_field
  attr_writer :write_only_field
  attr_accessor :shared_name

  private

  attr_accessor :private_field

  protected

  attr_reader :protected_field

  public

  def initialize
    @immutable_field = 1
  end
end

class Archive
  attr_accessor :shared_name
end
